src/reducers/index.ts

```diff
@@ -105,19 +105,31 @@ function parseVega(
   };
 }
 
-function parseVegaLite(
-  state: State,
-  action: SetVegaLiteExample | UpdateVegaLiteSpec | SetGistVegaLiteSpec,
-  extend: Partial<State> = {}
-) {
+function parseVegaLite(state: State, action: any, extend: Partial<State> = {}) {
   const currLogger = new LocalLogger();
 
   try {
+    // if we are setting the spec we don't pass the config
+    // add the previous config from the state
+    if (!action.config) {
+      action.config = state.config;
+    }
+
+    // if we are setting the config we don't pass the spec
+    // add the previous spec from the state
+    if (!action.spec) {
```

Comment:
same as above

Reply:
![image](https://user-images.githubusercontent.com/30566406/61211880-5c3efc80-a71e-11e9-8c90-861f8ecc7dcd.png)
We have to do this because of this error

Reply:
JSON.parse(undefined||"{}") throws an error

Reply:
Works for object though!